src/time-picker/TimePicker.js

```diff
@@ -136,13 +139,17 @@ export const filterTime = ({ originalInput, minTime, maxTime }) => time => {
   return lteTime(minTime, time) && lteTime(time, maxTime) && startsWith(time, originalInput);
 };
 
-export const makeOptions = ({ minTime, maxTime, timeFormat }, inputValue) => {
+export const makeOptions = ({ minTime, maxTime, timeFormat, userValue }, inputValue) => {
   const time = parseInTimeFormat(inputValue, timeFormat);
-  const timeList = time === inputError ? [] : createTimeList(time, timeFormat);
+  //make sure the userValue has timeFormat so in H12 the day period will be shown
+  if ( userValue && userValue !== inputError ) {
+    userValue.timeFormat = timeFormat;
+  }
```

Comment:
You're right!
Still, I don't like to modify function parameters. I'll leave a comment in the next review